mloop_multishot.py

```diff
@@ -13,6 +14,57 @@
 check_version('labscript_utils', '2.12.5', '3.0')
 
 
+def check_runmanager(config):
+    msgs = ['WARNING(s):']
+    rm_globals = rm.get_globals()
+    if not all([x in rm_globals for x in config['mloop_params']]):
+        msgs.append('Not all optimisation parameters present in runmanager.')
+    if not rm.get_run_shots():
+        msgs.append('Run shot(s) not selected in runmanager.')
+    if rm.error_in_globals():
+        msgs.append('Error in runmanager globals.')
+    n_shots = rm.n_shots()
+    if n_shots > 1 and not config['ignore_bad']:
+        msgs.append(
+            f'runmanager set to compile {n_shots:d} shots per request, but your '
+            + 'mloop_config has ignore_bad = False. You are advised to (i) remove '
+            + 'iterable globals so as to compile one shot per cost; or set ignore_bad '
+            + '= True in your mloop_config and only return one cost with bad = False '
+            + ' per sequence.'
+        )
+    if len(msgs) > 1:
+        print('\n'.join(msgs))
+    return len(msgs) <= 1
+
+
+def verify_globals(config):
+    # Get the current runmanager globals
+    rm_globals = rm.get_globals()
+    current_values = [rm_globals[name] for name in config['mloop_params']]
+
+    # Retrieve the parameter values requested by M-LOOP on this iteration
+    requested_values = lyse.routine_storage.params
+    requested_dict = dict(zip(config['mloop_params'], requested_values))
+
+    # Get the parameter values for the shot we just computed the cost for
+    df = lyse.data()
+    ix = -1
+    run = lyse.Run(df['filepath'].iloc[ix])
+    shot_globals = run.get_globals()
+    shot_values = [shot_globals[name] for name in config['mloop_params']]
+
+    # Verify integrity by cross-checking against what was requested
+    if current_values != requested_values:

```

Comment:
This is giving me
```python
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
Probably want:
```suggestion
    if not np.array_equal(current_values, requested_values):
```